spring-boot-project/spring-boot-docs/src/docs/asciidoc/features/external-config.adoc

```diff
@@ -608,6 +608,19 @@ The preceding example would be transformed into these properties:
 TIP: Properties that use the `[index]` notation can be bound to Java `List` or `Set` objects using Spring Boot's `Binder` class.
 For more details see the "`<<features#features.external-config.typesafe-configuration-properties>>`" section below.
 
+[[features.external-config.yaml.using-env-variables-in-yaml]]
+==== Using environment variables and default values in YAML property definitions
+
+In order to use environment variable as a source for application property, one can use placeholders like this: 
+
+[source,yaml,indent=0,subs="verbatim"]
+----
+	environments:
+	  dev:
+	    url: ${DEV_ENV_URL}
+	    name: ${DEV_ENV_NAME:development-environment}
```

Comment:
by the way, would be great to see how to use values with spaces, or "in quotes" as default values

Reply:
I don't think spaces in the default makes any difference. For example, you can do this:

```
property.example=${some.other.property:The default value}
```

Assuming that `some.other.property` hasn't been set the value of `property.example` will default to `The default value`.